builtin/tests/builtin.py

```diff
@@ -19,6 +19,10 @@
 assert any(["hello", "world"]) == True
 assert any([]) == False
 
+doc="ascii"
+assert ascii('hello world') == "'hello world'"
+assert ascii('안녕 세상') == "'\\uc548\\ub155 \\uc138\\uc0c1'"
```

Comment:
I think this needs a few more test cases, and you'll see that there are some cases missing from the above

```
>>> ascii("\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1A\x1B\x1C\x1D\x1E\x1F")
"'\\x00\\x01\\x02\\x03\\x04\\x05\\x06\\x07\\x08\\t\\n\\x0b\\x0c\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\\x14\\x15\\x16\\x17\\x18\\x19\\x1a\\x1b\\x1c\\x1d\\x1e\\x1f'"
>>> print(ascii("\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1A\x1B\x1C\x1D\x1E\x1F"))
'\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f'
>>> print(ascii("'hello'"))
"'hello'"
>>> print(ascii('"hello"'))
'"hello"'
>>> print(ascii('\\'))
'\\'
```

In fact you could do worse than adapting the tests here.

https://github.com/go-python/gpython/blob/bd0985be3562d0a36eef5e34ecae4d4a4235fd57/py/tests/string.py#L35

Note that this should probably call `ascii()` on some non strings too.